compat/win32_maperror.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/win32_maperror.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_ && !_SQUID_CYGWIN_
+
+#if HAVE_WINDOWS_H
+#include <windows.h>
+#endif
+// for _doserrno
+#if HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
+#include <unordered_map>
+
+void
+WIN32_maperror(unsigned long WIN32_oserrno)
+{
+    static const auto errormap = std::unordered_map<unsigned long, int> {
```

Comment:
> extracted WIN32_maperror into own header and implementation files using NoMoreGlobals

This is not a NoMoreGlobals pattern! This code will not work reliably if accessed after main(). Consult wiki and add `new`.

Reply:
Doh. Added. Thanks